libraries/lib-components/EffectInterface.h

```diff
@@ -433,17 +433,29 @@ class COMPONENTS_API EffectUIClientInterface /* not final */
    virtual bool IsGraphicalUI() = 0;
 
    //! Adds controls to a panel that is given as the parent window of `S`
-   virtual bool PopulateUI(ShuttleGui &S, EffectSettingsAccess &access) = 0;
+   /*!
+    @param S interface for adding controls to a panel in a dialog
+    @param access guaranteed to have a lifetime containing that of the returned
+    object
 
-   virtual bool ValidateUI() = 0;
-   virtual bool CloseUI() = 0;
+    @return null for failure; else an object invoked to retrieve values of UI
```

Comment:
I do not understand this comment: "return... an object invoked to retrieve values of UI controls" 
I look at the definition of EffectUIValidator and all I see there is `virtual bool Validate() = 0;` - is there a plan to extend that interface later on?

Reply:
When I said "retrieve" I mean the object is expected to respond to the `Validate` message by examining all controls and updating its state.  I did not mean that it returns anything to the caller besides the success code.  I could clarify the comments.